### ISSUE TITLE
Prefresh uses the port of the dev server regardless of where the content was served from

### ISSUE THREAD

**Issue by rschristian** [preactjs/preact-cli#1532]
**Do you want to request a _feature_ or report a _bug_?**

Bug

**What is the current behaviour?**

Currently writing an Ngrok alternative, and I've noticed Prefresh does not necessarily use the correct port for its communications. Ngrok, if you didn't know is a tool for allowing you to create HTTP tunnels so you could access a local dev server via a public URL. 

The issue is that, while all content is served correctly (meaning the page loads just fine), Prefresh will end up making requests to `https://public.url:3000`  (assuming my local dev server was running on port 3000), even if my public URL is `https://public.url`.

Probably a simple case of ensuring the plugin doesn't assume a port.

**What is the expected behaviour?**

That the request from Prefresh goes to the correct address

**Comment by JoviDeCroock** [comment]
I don't think that is an issue with prefresh as much as it is an issue with how the `web-dev-server`/`hot-plugin` is injected, a different port should be reflected on that I'd think.

**Comment by rschristian** [comment]
Fair enough, I currently know probably next to nothing about what Prefresh is actually doing, so I just took a guess before investigating further. Certainly will look at those then.

**Comment by JoviDeCroock** [comment]
I guess a good entry point would be to check whether or not normal hot-mode works i.e. if signals reach the client and if normal `preact dev` full reload mode works

**Comment by rschristian** [comment]
This looks to be a famously challenging problem (now that I've been pointed in the right direction, sorry Prefresh!), so a solution probably isn't going to be pleasant, just in case anyone else runs into this.
